### PR TITLE
LPS-205028 Fix error message on form container when guest user tries to add object entry beyond their limit

### PR DESCRIPTION
**Pull request by thallescezar**
Related issue: https://liferay.atlassian.net/browse/LPS-205028

### CODE REVIEW

**Comment by thallescezar** [comment]
ci:test:object

**Comment by thallescezar** [comment]
ci:test:relevant

File: modules/apps/info/info-api/src/main/java/com/liferay/info/exception/InfoFormValidationException.java [hunk 1]
```diff
@@ -57,6 +57,36 @@ public String getLocalizedMessage(Locale locale) {
...
+			super(StringPool.BLANK);
+
+			_objectDefinitionLabel = objectDefinitionLabel;
+		}
+
+		@Override
+		public String getLocalizedMessage(Locale locale) {
+			return LanguageUtil.format(
+				locale,
+				"the-limit-of-guest-entries-for-object-definition-has-been-" +
```
Comment (guilhermedcamacho) [inline modules/apps/info/info-api/src/main/java/com/liferay/info/exception/InfoFormValidationException.java:72]: I think you don't need to have this message key hard-coded here, you can pass the `ObjectEntryCountException` in your class constructor and use `getMessageKey()` method.

**Review by guilhermedcamacho — CHANGES_REQUESTED** [review]

**Comment by thallescezar** [comment]
ci:test:sf

File: modules/apps/info/info-api/src/main/java/com/liferay/info/exception/InfoFormValidationException.java [hunk 2]
```diff
@@ -57,6 +57,33 @@ public String getLocalizedMessage(Locale locale) {
 
 	}
 
+	public static class ExceedsMaxEntries extends InfoFormValidationException {
+
+		public ExceedsMaxEntries(
+			String messageKey, String objectDefinitionLabel) {
+
+			super(StringPool.BLANK);
```
Comment (guilhermedcamacho) [inline modules/apps/info/info-api/src/main/java/com/liferay/info/exception/InfoFormValidationException.java:65]: Is it really necessary?

File: modules/apps/info/info-api/src/main/java/com/liferay/info/exception/InfoFormValidationException.java [hunk 3]
```diff
@@ -57,6 +57,33 @@ public String getLocalizedMessage(Locale locale) {
...
+
+			_messageKey = messageKey;
+			_objectDefinitionLabel = objectDefinitionLabel;
+		}
+
+		@Override
+		public String getLocalizedMessage(Locale locale) {
+			return getLocalizedMessage(_objectDefinitionLabel, locale);
```
Comment (guilhermedcamacho) [inline modules/apps/info/info-api/src/main/java/com/liferay/info/exception/InfoFormValidationException.java:73]: We cannot pass `_objectDefinitionLabel` as `fieldLabel`.

File: modules/apps/info/info-api/src/main/java/com/liferay/info/exception/InfoFormValidationException.java [hunk 4]
```diff
@@ -57,6 +57,33 @@ public String getLocalizedMessage(Locale locale) {
...
+		}
+
+		@Override
+		public String getLocalizedMessage(String fieldLabel, Locale locale) {
+			return LanguageUtil.format(
```
Comment (guilhermedcamacho) [inline modules/apps/info/info-api/src/main/java/com/liferay/info/exception/InfoFormValidationException.java:78]: I think you should follow `InvalidCaptcha` pattern, so this logic should be moved to `getLocalizedMessage(Locale locale)` and you call this method here.

**Review by guilhermedcamacho — CHANGES_REQUESTED** [review]

**Comment by thallescezar** [comment]
ci:test:sf

File: modules/apps/info/info-api/src/main/java/com/liferay/info/exception/InfoFormValidationException.java [hunk 5]
```diff
@@ -57,6 +57,32 @@ public String getLocalizedMessage(Locale locale) {
...
+		@Override
+		public String getLocalizedMessage(Locale locale) {
+			return LanguageUtil.format(
+				locale, _messageKey, new String[] {_objectDefinitionLabel},
+				false);
+		}
+
+		@Override
+		public String getLocalizedMessage(String fieldLabel, Locale locale) {
+			return getLocalizedMessage();
```
Comment (guilhermedcamacho) [inline modules/apps/info/info-api/src/main/java/com/liferay/info/exception/InfoFormValidationException.java:78]: Hey @thallescezar , you should pass the locale here to call the method above.

**Review by guilhermedcamacho — CHANGES_REQUESTED** [review]

**Review by guilhermedcamacho — APPROVED** [review]
LGTM :tada:

**Comment by thallescezar** [comment]
Sent for Echo approval [here](https://github.com/liferay-echo/liferay-portal/pull/14973)

**Pull request closed without merging**